cmake/modules/configuration_files.cmake

```diff
@@ -67,6 +67,8 @@ elseif(CACHED_CONF_FILE)
   set(CONF_FILE ${CACHED_CONF_FILE})
 elseif(EXISTS   ${APPLICATION_CONFIG_DIR}/prj_${BOARD}.conf)
   set(CONF_FILE ${APPLICATION_CONFIG_DIR}/prj_${BOARD}.conf)
+  message(WARNING "Note that `prj_<board>.conf` files are deprecated and should be "
+                  "replaced with board Kconfig fragments instead.")
```

Comment:
minor nit, if we deprecate this, then en entry should be made here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/FindDeprecated.cmake

together with a comment on when this was deprecated.

This makes it easier to keep an overview of deprecated build system areas and thereby remove the code.
Also warning should be of type: `DEPRECATION` instead of `WARNING`.